api/include/opentelemetry/trace/propagation/http_trace_context.h

```diff
@@ -0,0 +1,278 @@
+// Example:
+//    HttpTraceContext.inject(setter,&carrier,&context);
+//    HttpTraceContext.extract(getter,&carrier,&context);
+template <typename T>
+class HttpTraceContext : public HTTPTextFormat<T>
+{
+public:
+  // Rules that manages how context will be extracted from carrier.
+  using Getter = nostd::string_view (*)(const T &carrier, nostd::string_view trace_type);
+
+  // Rules that manages how context will be injected to carrier.
+  using Setter = void (*)(T &carrier,
+                          nostd::string_view trace_type,
+                          nostd::string_view trace_description);
+
+  void Inject(Setter setter, T &carrier, const context::Context &context) noexcept override
+  {
+    SpanContext span_context = SpanContext();
+    GetCurrentSpan(context, span_context);
+    if (!span_context.IsValid())
+    {
+      return;
+    }
+    InjectImpl(setter, carrier, span_context);
+  }
+
+  context::Context Extract(Getter getter,
+                           const T &carrier,
+                           context::Context &context) noexcept override
+  {
+    SpanContext span_context    = ExtractImpl(getter, carrier);
+    nostd::string_view span_key = "current-span";
+    nostd::shared_ptr<Span> sp{new DefaultSpan(span_context)};
+    return context.SetValue(span_key, sp);
+  }
+
+  static void GetCurrentSpan(const context::Context &context, SpanContext &span_context)
+  {
+    const nostd::string_view span_key = "current-span";
+    context::Context ctx(context);
+    context::ContextValue span = ctx.GetValue(span_key);
+    if (nostd::holds_alternative<nostd::shared_ptr<Span>>(span))
+    {
+      span_context = nostd::get<nostd::shared_ptr<Span>>(span).get()->GetContext();
+    }
+  }
+
+  static TraceId GenerateTraceIdFromString(nostd::string_view trace_id)
+  {
+    const char *trc_id = trace_id.begin();
+    uint8_t buf[kTraceIdBytes / 2];
+    for (int i = 0; i < kTraceIdBytes; i++)
+    {
+      int tmp = CharToInt(*trc_id);
+      if (tmp < 0)
+      {
+        for (int j = 0; j < kTraceIdBytes / 2; j++)
+        {
+          buf[j] = 0;
+        }
+        return TraceId(buf);
+      }
+      if (i % 2 == 0)
+      {
+        buf[i / 2] = tmp * 16;
+      }
+      else
+      {
+        buf[i / 2] += tmp;
+      }
+      trc_id++;
+    }
+    return TraceId(buf);
+  }
+
+  static SpanId GenerateSpanIdFromString(nostd::string_view span_id)
+  {
+    const char *spn_id = span_id.begin();
+    uint8_t buf[kSpanIdBytes / 2];
+    for (int i = 0; i < kSpanIdBytes; i++)
+    {
+      int tmp = CharToInt(spn_id[i]);
+      if (tmp < 0)
+      {
+        for (int j = 0; j < kSpanIdBytes / 2; j++)
+        {
+          buf[j] = 0;
+        }
+        return SpanId(buf);
+      }
+      if (i % 2 == 0)
+      {
+        buf[i / 2] = tmp * 16;
+      }
+      else
+      {
+        buf[i / 2] += tmp;
+      }
+    }
+    return SpanId(buf);
+  }
+
+  static TraceFlags GenerateTraceFlagsFromString(nostd::string_view trace_flags)
```

Comment:
Although you might have a check before calling this method, you should still handle the case where `trace_flags` has more than 2 characters?

Reply:
I've change the code so that it will return an invalid Trace Flag upon a longer string